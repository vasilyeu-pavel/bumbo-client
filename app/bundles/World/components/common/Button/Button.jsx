import React, { PropTypes } from 'react';

const Button = ({ children, color, type, tag, modifiers, href, target, onClick }) => {
  let className = 'button';

  if (color) {
    className += ` ${`button--${color}`}`;
  }

  if (modifiers.length) {
    modifiers.map((modifier) => className += ` ${`button--${modifier}`}`);
  }

  const attributes = { className };

  if (onClick) attributes.onClick = onClick;

  if (type) attributes.type = type;

  if (href) {
    attributes.href = href;
    attributes.target = target;
  }

  return React.createElement(tag, attributes, children);
};

Button.defaultProps = {
  tag: 'button',
  modifiers: [],
};

Button.propTypes = {
  color: PropTypes.string,
  type: PropTypes.string,
  tag: PropTypes.string,
  modifiers: PropTypes.arrayOf(PropTypes.string),
  onClick: PropTypes.func,
  href: PropTypes.string,
  target: PropTypes.string,
};

export default Button;
