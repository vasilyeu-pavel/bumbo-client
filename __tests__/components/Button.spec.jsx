/* global expect, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../app/bundles/World/components/common/Button/Button';

describe('<Button /> component', () => {
  it('should render button tag by default', () => {
    const button = renderer.create(<Button />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render link tag', () => {
    const button = renderer.create(<Button tag='a' />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render link tag with href="/" and target="_blank"', () => {
    const button = renderer.create(<Button tag='a' href='/' target='_blank' />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render button with class "button--orange"', () => {
    const button = renderer.create(<Button color='orange' />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render button with class "button--big" and "button--linear"', () => {
    const button = renderer.create(<Button modifiers={['big', 'linear']} />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render button with text "Send"', () => {
    const text = 'Send';
    const button = renderer.create(<Button>{text}</Button>).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should render button with type "submit"', () => {
    const type = 'submit';
    const button = renderer.create(<Button type={type} />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should call "onClick" method after click on button', () => {
    const handler = jest.fn();
    const button = renderer.create(<Button onClick={handler} />).toJSON();

    button.props.onClick();

    expect(handler).toHaveBeenCalled();
  });
});
