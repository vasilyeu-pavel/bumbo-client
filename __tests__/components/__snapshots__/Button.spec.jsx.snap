exports[`<Button /> component should render button tag by default 1`] = `
<button
  className="button" />
`;

exports[`<Button /> component should render button with class "button--big" and "button--linear" 1`] = `
<button
  className="button button--big button--linear" />
`;

exports[`<Button /> component should render button with class "button--orange" 1`] = `
<button
  className="button button--orange" />
`;

exports[`<Button /> component should render button with text "Send" 1`] = `
<button
  className="button">
  Send
</button>
`;

exports[`<Button /> component should render button with type "submit" 1`] = `
<button
  className="button"
  type="submit" />
`;

exports[`<Button /> component should render link tag 1`] = `
<a
  className="button" />
`;

exports[`<Button /> component should render link tag with href="/" and target="_blank" 1`] = `
<a
  className="button"
  href="/"
  target="_blank" />
`;
